src/RwCalculators/GReWeighthA2025.cxx

```diff
@@ -0,0 +1,299 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2018, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+ Authors: Costas Andreopoulos <costas.andreopoulos \at stfc.ac.uk>
+          University of Liverpool & STFC Rutherford Appleton Lab
+
+          Jim Dobson <J.Dobson07 \at imperial.ac.uk>
+          Imperial College London
+*/
+//____________________________________________________________________________
+
+#include <cassert>
+#include <cstdlib>
+
+#include <TMath.h>
+#include <TNtuple.h>
+#include <TFile.h>
+#include <TLorentzVector.h>
+#include <TVector.h>
+
+// GENIE/Generator includes
+#include "Framework/Algorithm/AlgFactory.h"
+#include "Framework/Algorithm/AlgConfigPool.h"
+#include "Framework/Conventions/Units.h"
+#include "Framework/EventGen/EventRecord.h"
+#include "Framework/GHEP/GHepParticle.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/Numerical/Spline.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/Registry/Registry.h"
+#include "Physics/NuclearState/NuclearUtils.h"
+#include "Physics/HadronTransport/HAIntranuke2018.h"
+#include "Physics/HadronTransport/Intranuke2018.h"
+#include "Physics/HadronTransport/INukeHadroData2018.h"
+#include "Physics/HadronTransport/INukeHadroFates2018.h"
+#include "Physics/HadronTransport/INukeUtils2018.h"
+
+#include "Physics/HadronTransport/INukeHadroData2025.h"
+#include "Physics/HadronTransport/INukeHadroFates2025.h"
+#include "Physics/HadronTransport/INukeUtils2025.h"
+#include "Physics/HadronTransport/HAIntranuke2025.h"
+#include "Physics/HadronTransport/Intranuke2025.h"
+
+// GENIE/Reweight includes
+#include "RwCalculators/GReWeighthA2025.h"
+#include "RwCalculators/GReWeightUtils.h"
+#include "RwFramework/GSystUncertainty.h"
+
+using namespace genie;
+using namespace genie::rew;
+
+//_______________________________________________________________________________________
+GReWeighthA2025::GReWeighthA2025() :
+GReWeightModel("IntraNuke")
+{
+#ifdef _G_REWEIGHT_INUKE_DEBUG_NTP_
+  fTestFile = new TFile("./intranuke_reweight_test.root","recreate");
+  fTestNtp  = new TNtuple("testntp","","pdg:E:mfp_twk_dial:d:d_mfp:fate:interact:w_mfp:w_fate");
+#endif
+
+  // Look up the FSI model for the current tune. Also check whether FSIs are
+  // actually enabled.
+  AlgConfigPool* conf_pool = AlgConfigPool::Instance();
+  Registry* gpl = conf_pool->GlobalParameterList();
+  RgAlg fsi_alg = gpl->GetAlg( "HadronTransp-Model" );
+  bool fsi_enabled = gpl->GetBool( "HadronTransp-Enable" );
+
+  if ( !fsi_enabled ) {
+    LOG( "ReW", pERROR ) << "FSIs are not enabled for the current tune."
+      << " Refusing to reweight FSIs.";
+    std::exit( 1 );
```

Comment:
If the code is going to exit(1) then the LOG message should be using pFATAL level.